src/js/main.js

```diff
@@ -63,131 +75,128 @@ var SupportKit = Marionette.Object.extend({
+                    $('body').append(link);
+                });
+
+                this.ready = true;
+                return Promise.resolve();
+            }
+
+            // TODO: alternatively load fallback CSS that doesn't use
+            // unsupported things like transforms
+            if (!$.support.cssProperty('transform')) {
+                return Promise.reject(new Error('SupportKit is not supported on this browser. ' +
+                        'Missing capability: css-transform'));
+            }
+
 
+            this.ready = false;
+            options = options || {};
 
-        if (/lebo|awle|pide|obo|rawli/i.test(navigator.userAgent)) {
-            var link = $('<a>')
-                .attr('href', 'https://supportkit.io?utm_source=widget')
-                .text('In app messaging by supportkit');
+            // if the email was passed at init, it can't be changed through the web widget UI
+            var readOnlyEmail = !_.isEmpty(options.email);
+            var emailCaptureEnabled = options.emailCaptureEnabled && !readOnlyEmail;
+            var uiText = _.extend({}, this.defaultText, options.customText);
 
-            $(function() {
-                $('body').append(link);
+            this.options = _.defaults(options, {
+                emailCaptureEnabled: emailCaptureEnabled,
+                readOnlyEmail: readOnlyEmail,
+                uiText: uiText
             });
 
-            this.ready = true;
-            return;
-        }
+            if (typeof options === 'string') {
+                options = {
+                    appToken: options
+                };
+            }
 
-        // TODO: alternatively load fallback CSS that doesn't use
-        // unsupported things like transforms
-        if (!$.support.cssProperty('transform')) {
-            console.error('SupportKit is not supported on this browser. ' +
-                'Missing capability: css-transform');
-            return;
-        }
+            if (typeof options === 'object') {
+                endpoint.appToken = options.appToken;
+                if (options.serviceUrl) {
+                    endpoint.rootUrl = options.serviceUrl;
+                }
+            } else {
+                return Promise.reject(new Error('init method accepts an object or string'));
+            }
 
-        this.ready = false;
-        options = options || {};
+            if (!endpoint.appToken) {
+                return Promise.reject(new Error('init method requires an appToken'));
+            }
 
-        options = _.defaults(options, {
-            emailCaptureEnabled: false
-        });
+            return this.login(options.userId, options.jwt);
+    },
 
-        if (typeof options === 'object') {
-            endpoint.appToken = options.appToken;
-            endpoint.jwt = options.jwt;
-            if (options.serviceUrl) {
-                endpoint.rootUrl = options.serviceUrl;
+    login: function(userId, jwt) {
```

Comment:
When a user transitions from no userId to having a userId, we should clear the last read timestamp for the anonymous user, since they will be upgraded into the new `userId` and the conversation will be cleared